### PR TITLE
Meta+Toolchain+Userland+AK: Bring Jakt in as part of the toolchain!

### PR DESCRIPTION
**Pull request by alimpfard**
Continuation of #20765

### CODE REVIEW

**Comment by alimpfard** [comment]
Commits are slightly messed-up, will fix.

File: Meta/CMake/common_options.cmake [hunk 1]
```diff
@@ -26,7 +26,7 @@ serenity_option(ENABLE_ACCELERATED_GRAPHICS ON CACHE BOOL "Enable use of acceler
 
 serenity_option(HACKSTUDIO_BUILD OFF CACHE BOOL "Automatically enabled when building from HackStudio")
 
-serenity_option(ENABLE_JAKT OFF CACHE BOOL "Enable building jakt files")
+serenity_option(ENABLE_JAKT ON CACHE BOOL "Enable building jakt files")
```
Comment (alimpfard) [inline Meta/CMake/common_options.cmake:29]: TODO: Still need to figure out how to deal with this being set to OFF.

File: Meta/CMake/utils.cmake [hunk 2]
```diff
@@ -126,7 +156,6 @@ endfunction()
 function(serenity_app target_name)
     cmake_parse_arguments(PARSE_ARGV 1 SERENITY_APP "" "ICON" "")
 
-    serenity_bin("${target_name}")
     set(small_icon "${SerenityOS_SOURCE_DIR}/Base/res/icons/16x16/${SERENITY_APP_ICON}.png")
     set(medium_icon "${SerenityOS_SOURCE_DIR}/Base/res/icons/32x32/${SERENITY_APP_ICON}.png")
```
Comment (alimpfard) [inline Meta/CMake/utils.cmake:160]: TODO: Icons seem to be broken with jakt apps.

File: Userland/Applications/Weather/weather.jakt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    temperature: Temperature
+    icon: Icon
+    timestamp: i64
+}
+
+trait Backend {
+    fn search(this, query: String) throws -> WeatherData
+    fn search(this, coords: Coords) throws -> WeatherData
+    fn forecast(this, data: &WeatherData) throws -> Forecast
+}
```
Comment (alimpfard) [inline Userland/Applications/Weather/weather.jakt:114]: TODO: Don't need this, remove it.

File: Userland/Libraries/LibC/CMakeLists.txt [hunk 4]
```diff
@@ -302,7 +302,7 @@ target_link_libraries(DynamicLoader_LibC
 # FIXME: Remove this on the next GCC toolchain version bump
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     add_custom_target(create_compat_ssp_libraries)
-    add_dependencies(Loader.so create_compat_ssp_libraries)
+    add_dependencies(LibC create_compat_ssp_libraries)
```
Comment (DanShaders) [inline Userland/Libraries/LibC/CMakeLists.txt:305]: Why?

Comment (alimpfard) [reply Userland/Libraries/LibC/CMakeLists.txt:305]: Not sure, it was a while ago.
I had build failures with LibC needing some ssp whatever on rebuilds, this fixed it :shrug:

Comment (alimpfard) [reply Userland/Libraries/LibC/CMakeLists.txt]: I dropped the change, not sure if we need it - will mention if I see a breakage.

File: Meta/CMake/serenity_compile_options.cmake [hunk 5]
```diff
@@ -26,8 +26,6 @@ add_compile_options(-fstack-clash-protection)
 add_compile_options(-fstack-protector-strong)
 add_link_options(-fstack-protector-strong)
 
-add_compile_options(-g1)
-
```
Comment (alimpfard) [inline Meta/CMake/serenity_compile_options.cmake:30]: TODO: This should not be in this commit.

File: Meta/CMake/common_options.cmake (hunk 1 above)
Comment (alimpfard) [reply Meta/CMake/common_options.cmake:29]: CMake bullshit'd it away.

File: Toolchain/BuildJakt.sh [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+                    -e "s|$SRC_ROOT/Kernel/|Kernel/|" \
+                    -e "s|$SRC_ROOT/Userland/Libraries/LibELF/|LibELF/|" \
+                    -e "s|$SRC_ROOT/Userland/Libraries/LibRegex/|LibRegex/|")
+                buildstep "system_headers" mkdir -p "$(dirname "Root/usr/include/$target")"
+                buildstep "system_headers" "$INSTALL" "$header" "Root/usr/include/$target"
+            done
+            unset SRC_ROOT
+        popd
+
+        echo "XXX build jakt support libs"
```
Comment (nico) [inline Toolchain/BuildJakt.sh:218]: XXX

**Comment by alimpfard** [comment]
YOLO.

**Pull request merged**